library/core/src/backtrace.rs

```diff
@@ -0,0 +1,231 @@
+//! hi
+#![unstable(feature = "core_backtrace", issue = "74465")]
+use crate::{fmt, mem, ptr};
+
+/// The current status of a backtrace, indicating whether it was captured or
+/// whether it is empty for some other reason.
+#[non_exhaustive]
+#[derive(Debug, PartialEq, Eq)]
+pub enum BacktraceStatus {
+    /// Capturing a backtrace is not supported, likely because it's not
+    /// implemented for the current platform.
+    Unsupported,
+    /// Capturing a backtrace has been disabled through either the
+    /// `RUST_LIB_BACKTRACE` or `RUST_BACKTRACE` environment variables.
+    Disabled,
+    /// A backtrace has been captured and the `Backtrace` should print
+    /// reasonable information when rendered.
+    Captured,
+}
+
+/// Global implementation of backtrace functinality. Called to create
+/// `RawBacktrace` trait objects.
+#[allow(improper_ctypes)]
+extern {
+    #[linkage = "extern_weak"]
+    #[link_name = "__rust_backtrace_capture"]
+    static BACKTRACE_CAPTURE: *mut ();
+
+    #[linkage = "extern_weak"]
```

Comment:
1. Is weak linkage supported by all object file formats that we would ever want to support? I guess not.
2. I think trying generate a backtrace when this isn't possible should give a compile time error. Otherwise you may not realize that backtraces are unavailable until you need them.

Reply:
1. not from what I've been told
2. :+1:

I've since swapped to using a `weak_lang_item` which i think gives the property from (2)